src/lib/definition-parser.ts

```diff
@@ -107,50 +116,58 @@ export async function getTypingInfo(folderName: string): Promise<TypingParseFail
 		mi.declaredModules.push(folderName);
 	}
 
-	function regexMatch(rx: RegExp, defaultValue: string): string {
-		const match = rx.exec(entryPointContent);
-		return match ? match[1] : defaultValue;
+	return Object.assign({fileKind}, mi);
+}
+
+export async function getTypingInfo(folderName: string): Promise<TypingParseFailResult | TypingParseSucceedResult> {
+	const [log, logResult] = quietLoggerWithErrors();
+	const directory = definitelyTypedPath(folderName);
+	if (folderName !== folderName.toLowerCase()) {
+		throw new Error(`Package name \`${folderName}\` should be strictly lowercase`);
 	}
 
-	const authors = regexMatch(/^\/\/ Definitions by: (.+)$/m, "Unknown");
-	const libraryMajorVersion = regexMatch(/^\/\/ Type definitions for [^\n]+ v?(\d+)/m, "0");
-	const libraryMinorVersion = regexMatch(/^\/\/ Type definitions for [^\n]+ v?\d+\.(\d+)/m, "0");
-	const libraryName = regexMatch(/^\/\/ Type definitions for (.+)$/m, "Unknown").trim();
-	const projectName = regexMatch(/^\/\/ Project: (.+)$/m, "");
-	const packageName = path.basename(directory);
-	const sourceRepoURL = "https://www.github.com/DefinitelyTyped/DefinitelyTyped";
+	log.info(`Reading contents of ${directory}`);
 
-	if (packageName !== packageName.toLowerCase()) {
-		log.error(`Package name \`${packageName}\` should be strictly lowercase`);
+	// There is a *single* entry point, containing metadata comments.
+	// But there may be many entryFilenames, which are the starting points of inferring all files to be included.
+	const entryPointResult = await entryPoint(directory, folderName, log.info);
+	if (entryPointResult.kind === "failure") {
+		log.info(entryPointResult.message);
+		log.error(entryPointResult.message);
+		return { kind: "fail", logs: logResult(), rejectionReason: RejectionReason.TooManyFiles };

```

Comment:
may be we should call this something else other than entryPoint. maybe something like mainFile, packageFile, index.